modules/event-objects/src/main/java/com/intuit/wasabi/eventobjects/EventEnvelopePayload.java

```diff
@@ -88,18 +88,25 @@ public String toJson() {
         JSONObject eventJson = new JSONObject();
 
         eventJson.put("messageType", MessageType.EVENT.toString());
-        eventJson.put("applicationName", applicationName.toString());
-        eventJson.put("experimentLabel", experimentLabel.toString());
-        eventJson.put("userID", assignment.getUserID().toString());
-        eventJson.put("bucketLabel", assignment.getBucketLabel().toString());
+        eventJson.put("applicationName", applicationName != null ? applicationName.toString(): null);
+        eventJson.put("experimentLabel", experimentLabel != null ? experimentLabel.toString(): null);
+        eventJson.put("userID",
+                assignment != null && assignment.getUserID() != null ? assignment.getUserID().toString(): null);
+        eventJson.put("bucketLabel",
+                assignment != null && assignment.getBucketLabel() != null ?
+                        assignment.getBucketLabel().toString(): null);
         eventJson.put("time_uuid", makeUUID().toString());
-        eventJson.put("experimentID", assignment.getExperimentID().toString());
-        eventJson.put("context", assignment.getContext().toString());
-        eventJson.put("epochTimestamp", event.getTimestamp().getTime());
-        eventJson.put("eventType", event.getType() + "");
-        eventJson.put("eventName", event.getName() + "");
-        eventJson.put("eventPayload", event.getPayload());
-        eventJson.put("value", event.getValue());
+        eventJson.put("experimentID",
+                assignment != null && assignment.getExperimentID() != null ?
+                        assignment.getExperimentID().toString(): null);
+        eventJson.put("context",
+                assignment != null && assignment.getContext() != null ? assignment.getContext().toString(): null);
+        eventJson.put("epochTimestamp",
+                event != null && event.getTimestamp() != null ? event.getTimestamp().getTime(): null);
+        eventJson.put("eventType", event != null && event.getType() != null ? event.getType() + "": "null");
```

Comment:
Minor: You probably don't need "&& event.getType() != null" & "&& event.getName() != null"